### PR TITLE
Parser Config

### PR DESCRIPTION
**Pull request by garunagiri**

### CODE REVIEW

File: app/objects/c_parserconfig.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        self._validate()
+
+    def _validate(self):
+        if (self.edge is None) ^ (self.target is None):
+            raise ParserConfigException('Edge or Target provided without the other.')
+
+
+class ParserConfigException(Exception):
```
Comment (wbooth) [inline app/objects/c_parserconfig.py]: put this at the top

File: app/service/data_svc.py [hunk 2]
```diff
@@ -180,14 +181,14 @@ def __init__(self):
...
+                              platform, cleanup=None, payload=None, parsers=None, requirements=None, privilege=None):
         ps = []
         for module in parsers:
-            relation = [Relationship(source=r['source'], edge=r.get('edge'), target=r.get('target')) for r in
-                        parsers[module]]
-            ps.append(Parser(module=module, relationships=relation))
+            pcs = []
+            for m in parsers[module]:
+                pcs.append(ParserConfig(**m))
+            ps.append(Parser(module=module, parserconfigs=pcs))
```
Comment (wbooth) [inline app/service/data_svc.py]: can this be compressed into a list comprehension?

Comment (wbooth) [reply app/service/data_svc.py]: and be a one-liner

**Comment by garunagiri** [comment]
I messed up.

**Pull request closed without merging**